+neuron/Object.m

```diff
@@ -0,0 +1,163 @@
+        % Call method by passing method name (method) to HOC lookup, along with its return type (returntype) and method arguments (varargin).
+        %   value = call_method_double(method, varargin)
+            
+            try
+                n = length(varargin);
+                for i=1:n
+                    neuron.hoc_push(varargin{i});
+                end
+                sym = clib.neuron.hoc_table_lookup(method, ...
+                    self.obj.ctemplate.symtable);
+                clib.neuron.hoc_call_ob_proc(self.obj, sym, n);
+                value = neuron.hoc_pop(returntype);
+            % TODO: if the above code fails, Matlab often just crashes instead of catching an error.
+            catch  
+                warning("'"+string(method)+"': number or type of arguments incorrect.")
+            end
+
+        end
+
+        function list_methods(self)
+        % List all available methods to be called using HOC lookup.
+        %   list_methods()
+            if ~isempty(self.attr_list)
+                disp("Available attributes:")
+                for i=1:length(self.attr_list)
+                    disp("    "+self.attr_list(i));
+                end
+            end
+            if ~isempty(self.mt_double_list)
+                disp("Available methods (returning a double):")
+                for i=1:length(self.mt_double_list)
+                    disp("    "+self.mt_double_list(i));
+                end
+            end
+            if ~isempty(self.mt_object_list)
+                disp("Available methods (returning an object):")
+                for i=1:length(self.mt_object_list)
+                    disp("    "+self.mt_object_list(i));
+                end
+            end
+            if ~isempty(self.mt_string_list)
+                disp("Available methods (returning a string):")
+                for i=1:length(self.mt_string_list)
+                    disp("    "+self.mt_string_list(i));
+                end
+            end
+        end
+
+        function varargout = subsref(self, S)
+        % If a method is called, but it is not listed above, try to run it by calling self.call_method_hoc().
+
+            % S(1).subs is method name;
+            % S(2).subs is a cell array containing arguments.
+            method = S(1).subs;
+
+            % Is the provided method listed above?
+            if any(strcmp(methods(self), method))
+                [varargout{1:nargout}] = builtin('subsref', self, S);
+            % Are we trying to directly access a Neuron class property?
+            elseif (isa(method, "char") && length(S) == 1 && any(strcmp(self.attr_list, method)))
+                [varargout{1:nargout}] = clib.neuron.get_pp_property(self.obj, method);
+            % Are we trying to directly access a Matlab class property?
+            elseif (isa(method, "char") && length(S) == 1 && any(strcmp(properties(self), method)))
+                [varargout{1:nargout}] = self.(method);
```

Comment:
if also a GetMethod is added to the dynamic properties, could we then combine these two? I.e. are the dynamic properties also part of properties(self)?
Can then also use isprop() instead of the any(strcmp(properties()))?

Reply:
Yes! Fixed.